### ISSUE TITLE
Waiting for EBS volume is not reliable

### ISSUE THREAD

**Issue by a1exsh** [zalando-stups/taupage#322]
We've seen that sometimes Taupage doesn't wait long enough for the just attached EBS volume to become ready and the following mount attempt results in kernel error, which leads Taupage to believe the volume has no filesystem and needs to be formatted.

See https://github.com/zalando/planb-cassandra/issues/93 for kernel message example.

**Comment by a1exsh** [comment]
After merging #323 we now see the following errors in the syslog of testing instance:
```
taupage-init: WARNING: More than one EBS volume with name taupage-ami-test-vol1 found.
taupage-init: INFO: Attached EBS volume 'taupage-ami-test-vol1' as '/dev/xvdf'
taupage-init: WARNING: More than one EBS volume with name taupage-ami-test-vol2 found.
taupage-init: INFO: Attached EBS volume 'taupage-ami-test-vol2' as '/dev/xvdg'
taupage-init: mdadm: cannot open /dev/md/sampleraid0: No such file or directory
taupage-init: INFO: Waiting for /dev/xvdf to stabilize..
taupage-init: mdadm: cannot open /dev/xvdf: Device or resource busy
taupage-init: Traceback (most recent call last):
taupage-init:   File "./init.d/10-prepare-disks.py", line 268, in <module>
taupage-init:     main()
taupage-init:   File "./init.d/10-prepare-disks.py", line 261, in main
taupage-init:     handle_volumes(args, config)
taupage-init:   File "./init.d/10-prepare-disks.py", line 235, in handle_volumes
taupage-init:     handle_raid_volumes(volumes.get("raid"))
taupage-init:   File "./init.d/10-prepare-disks.py", line 222, in handle_raid_volumes
taupage-init:     create_raid_device(raid_device, raid_config)
taupage-init:   File "./init.d/10-prepare-disks.py", line 213, in create_raid_device
taupage-init:     subprocess.check_call(call)
taupage-init:   File "/usr/lib/python3.4/subprocess.py", line 561, in check_call
taupage-init:     raise CalledProcessError(retcode, cmd)
taupage-init: subprocess.CalledProcessError: Command '['mdadm', '--create', '/dev/md/sampleraid0', '--run', '--level=1', '--raid-devices=2', '/dev/xvdf', '/dev/xvdg']' returned non-zero exit status 2
taupage-init: ERROR: Failed to start 10-prepare-disks.py
```

I don't see how the actual change in that PR could cause this failure.  However, maybe we should not only wait for the device file to appear in `/dev` filesystem, but also verify that opening it doesn't cause error?

**Comment by a1exsh** [comment]
See #329 for possible fix.

**Comment by a1exsh** [comment]
Now that I'm thinking about it, is that the logic is still broken.  We should not just try to create filesystem on a partition we couldn't mount for whatever reason.

What we should do is tell apart retry-able and non-retry-able failures.  For the former ones we just retry enough times, possibly involving fsck, and if all fails we must assume an empty volume (or your data is lost either way).

**Comment by a1exsh** [comment]
The latest attempt to fix this is #342
